src/Workspaces/CSharp/Portable/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -300,6 +297,17 @@ private bool ShouldFormat()
                 return false;
             }
 
+            private bool OnDisabledTextTrivia(SyntaxTrivia trivia, int index)
+            {
+                if (trivia.IsKind(SyntaxKind.DisabledTextTrivia) &&
+                    SyntaxFacts.IsNewLine(trivia.ToString().Last()))
+                {
+                    ResetStateAfterNewLine(index);
+                }
+
+                return false;
```

Comment:
`OnDisabledTextTrivia()` like the other methods in `ShouldFormat()` is a method which introduces side effects. This is the intended behavior of the method.